src/components/DownloadManifestModal.js

```diff
@@ -30,103 +29,153 @@ const Button = compose(withTheme)(({ theme, children, ...props }) => (
+      & .copyContent {
+        padding-left: 20px;
+        padding-right: 20px;
+        flex: 1;
+        justify-content: center;
+        align-items: center;
+        display: flex;
+      }
+      & .generateButton {
+        border-top-left-radius: 0px;
+        border-bottom-left-radius: 0px;
+      }
+    }
+  `}`;
+
+const GenerateManifestSet = compose(injectState, withTheme)(
   ({
     api,
+    theme,
     setId,
     setSetId,
     sqon,
     setWarning,
     onManifestGenerated,
     state: { loggedInUser },
     effects: { addUserSet },
-  }) => (
-    <div>
-      {setId ? (
-        <CopyToClipboard value={setId} />
-      ) : (
+    className,
+  }) => {
+    const copyRef = React.createRef();
+    return (
+      <div className={`${ManifestGeneratorStyle()}`}>
         <LoadingOnClick
           onClick={async () => {
-            const type = 'file';
-            const { data, errors } = await saveSet({
-              type,
-              sqon: sqon || {},
-              userId: loggedInUser.egoId,
-              path: 'kf_id',
-              api: graphql(api),
-            });
-            if (errors && errors.length) {
-              setWarning('Unable to generate KF-get ID, please try again later.');
+            if (setId) {
+              copyValueToClipboard({ value: setId, copyRef });
             } else {
-              const { setId, size } = data.saveSet;
-              setWarning('');
-              setSetId(setId);
-              onManifestGenerated(data.saveSet);
-              addUserSet({ type, setId, size, api });
+              const type = 'file';
+              const { data, errors } = await saveSet({
+                type,
+                sqon: sqon || {},
+                userId: loggedInUser.egoId,
+                path: 'kf_id',
+                api: graphql(api),
+              });
+              if (errors && errors.length) {
+                setWarning('Unable to generate KF-get ID, please try again later.');
+              } else {
+                const { setId: receivedSetId, size } = data.saveSet;
+                setWarning('');
+                setSetId(receivedSetId);
+                onManifestGenerated(data.saveSet);
+                addUserSet({ type, setId: receivedSetId, size, api });
+              }
             }
           }}
           render={({ onClick, loading }) => (
-            <Button {...{ onClick, disabled: loading }}>
-              {loading && (
-                <Spinner
-                  fadeIn="none"
-                  name="circle"
-                  color="#fff"
-                  style={{
-                    width: 15,
-                    height: 15,
-                    marginRight: 9,
-                  }}
-                />
-              )}
-              <Trans>Generate KF-get ID</Trans>
-            </Button>
+            <Fragment>
+              <span ref={copyRef} className={`copyContent`}>
+                {setId || <Trans>Generate KF-get ID</Trans>}
```

Comment:
Is KF-get still the name for this? Thought the design may have included something different

Reply:
very nice catch! indeed it says `Generate manifest ID` :)